include/s3select.h

```diff
@@ -2329,13 +2384,17 @@ class base_s3object
           if (m_is_to_aggregate)
             for (auto& i : m_projections)
             {
+              if(i->is_column_reference())  {
```

Comment:
it is not correct to place `is_column_reference` under  `(m_aggr_flow == true)`.
upon aggregation flow the JSON output can not contain column reference